dpctl/tensor/libtensor/source/copy_numpy_ndarray_into_usm_ndarray.cpp

```diff
@@ -116,21 +116,29 @@ void copy_numpy_ndarray_into_usm_ndarray(
 
     // check for applicability of special cases:
     //      (same type && (both C-contiguous || both F-contiguous)
-    bool both_c_contig =
+    const bool both_c_contig =
         ((src_flags & py::array::c_style) && dst.is_c_contiguous());
-    bool both_f_contig =
+    const bool both_f_contig =
         ((src_flags & py::array::f_style) && dst.is_f_contiguous());
+
+    const bool same_data_types = (src_type_id == dst_type_id);
+
     if (both_c_contig || both_f_contig) {
-        if (src_type_id == dst_type_id) {
+        if (same_data_types) {
             int src_elem_size = npy_src.itemsize();
 
             sycl::event copy_ev =
                 exec_q.memcpy(static_cast<void *>(dst_data),
                               static_cast<const void *>(src_data),
                               src_nelems * src_elem_size, depends);
 
-            // wait for copy_ev to complete
-            copy_ev.wait_and_throw();
+            {
+                // wait for copy_ev to complete
+                // release GIL to allow other threads (host_tasks)
+                // a chance to acquire GIL
+                py::gil_scoped_release lock{};
+                copy_ev.wait_and_throw();
+            }
```

Comment:
Do we have tests for these cases? If not, would be nice to add a few.

Reply:
An example that hung due to unreleased GIL, discovered by @vtavana, was (with latest `dpnp`)

```python
import dpctl.tensor as dpt
import dpnp

a_py = [-1,2,1,3,-4]
a = dpnp.asarray(a_py, dtype=dpnp.complex64)
res = dpnp.fft.fft(a)
print("1")

a_usm = dpt.asarray(a, dtype="c8")
rest = dpnp.fft.fft(a_usm)
print("2")
``` 

It hung during the last call to `fft`, during call to `dpctl` to copy from `numpy.ndarray` to `usm_ndarray`.

I am yet unable to build a stand-alone `dpctl`-pure reproducer.